crates/settlement-clients/ethereum/src/lib.rs

```diff
@@ -1,87 +1,135 @@
 
 pub const ENV_PRIVATE_KEY: &str = "ETHEREUM_PRIVATE_KEY";
 
 lazy_static! {
     pub static ref CURRENT_PATH: PathBuf = std::env::current_dir().unwrap();
-    pub static ref KZG_SETTINGS: KzgSettings = KzgSettings::load_trusted_setup_file(
-        CURRENT_PATH.join("../../../orchestrator/src/jobs/state_update_job/trusted_setup.txt").as_path()
-    )
-    .expect("Error loading trusted setup file");
+    pub static ref KZG_SETTINGS: KzgSettings =
+        // TODO: set more generalized path
+        KzgSettings::load_trusted_setup_file(CURRENT_PATH.join("src/trusted_setup.txt").as_path())
+            .expect("Error loading trusted setup file");
 }
 
 #[allow(dead_code)]
 pub struct EthereumSettlementClient {
-    provider: Arc<EthHttpProvider>,
     core_contract_client: StarknetValidityContractClient,
     wallet: EthereumWallet,
     wallet_address: Address,
+    #[cfg(not(test))]
+    provider: Arc<EthHttpProvider>,
+    #[cfg(test)]
+    provider: RootProvider<Http<Client>>,
 }
 
 impl EthereumSettlementClient {
+    #[cfg(not(test))]
     pub fn with_settings(settings: &impl SettingsProvider) -> Self {
         let settlement_cfg: EthereumSettlementConfig = settings.get_settings(SETTLEMENT_SETTINGS_NAME).unwrap();
 
         let private_key = get_env_var_or_panic(ENV_PRIVATE_KEY);
         let signer: PrivateKeySigner = private_key.parse().expect("Failed to parse private key");
-        let wallet = EthereumWallet::from(signer.clone());
-
         let wallet_address = signer.address();
+        let wallet = EthereumWallet::from(signer);
 
         let provider = Arc::new(
             ProviderBuilder::new().with_recommended_fillers().wallet(wallet.clone()).on_http(settlement_cfg.rpc_url),
         );
         let core_contract_client = StarknetValidityContractClient::new(
-            Address::from_str(&settlement_cfg.core_contract_address)
-                .expect("Failed to convert the validity contract address.")
-                .0
-                .into(),
+            Address::from_str(&settlement_cfg.core_contract_address).unwrap().0.into(),
             provider.clone(),
         );
 
         EthereumSettlementClient { provider, core_contract_client, wallet, wallet_address }
     }
 
+    #[cfg(test)]
+    pub fn with_test_settings(settings: &impl SettingsProvider, provider: RootProvider<Http<Client>>) -> Self {
+        let settlement_cfg: EthereumSettlementConfig = settings.get_settings(SETTLEMENT_SETTINGS_NAME).unwrap();
+
+        let private_key = get_env_var_or_panic(ENV_PRIVATE_KEY);
+        let signer: PrivateKeySigner = private_key.parse().expect("Failed to parse private key");
+        let wallet_address = signer.address();
+        let wallet = EthereumWallet::from(signer);
+
+        let fill_provider = Arc::new(
+            ProviderBuilder::new().with_recommended_fillers().wallet(wallet.clone()).on_http(settlement_cfg.rpc_url),
+        );
+
+        let core_contract_address = if *SHOULD_IMPERSONATE_ACCOUNT {
```

Comment:
can you add a small comment explaining the rationale behind this

Reply:
Added.